Sources/GoTrue/GoTrueClient.swift

```diff
@@ -142,6 +142,25 @@ public final class GoTrueClient {
     return url
   }
 
+  public func refreshSession(refreshToken: String) async throws -> Session {
+    do {
+      let session = try await Current.client.send(
+        Paths.token.post(
+          grantType: .refreshToken,
+          .userCredentials(UserCredentials(refreshToken: refreshToken))
+        )).value
+      
+      if session.user.phoneConfirmedAt != nil {
```

Comment:
The session is updated only if the user created his account with the phone number. It will not work if the user signed up via email. So the condition should be like this to make it work:
```
session.user.phoneConfirmedAt != nil || session.user.emailConfirmedAt != nil || session.user.confirmedAt != nil
```